test/e2e/common/node/framework/cgroups/cgroups.go

```diff
@@ -0,0 +1,421 @@
+		roundedCPULimit := (milliCPULimit/10 + 1) * 10
+		cpuQuotaRounded := kubecm.MilliCPUToQuota(roundedCPULimit, kubecm.QuotaPeriod)
+		expectedCPULimits = append(expectedCPULimits, getExpectedCPULimitFromCPUQuota(cpuQuotaRounded, podOnCgroupV2))
+	}
+
+	return expectedCPULimits
+}
+
+func getExpectedCPULimitFromCPUQuota(cpuQuota int64, podOnCgroupV2 bool) string {
+	expectedCPULimitString := strconv.FormatInt(cpuQuota, 10)
+	if podOnCgroupV2 {
+		if expectedCPULimitString == "-1" {
+			expectedCPULimitString = "max"
+		}
+		expectedCPULimitString = fmt.Sprintf("%s %d", expectedCPULimitString, kubecm.QuotaPeriod)
+	}
+	return expectedCPULimitString
+}
+
+func getExpectedCPUShares(rr *v1.ResourceRequirements, podOnCgroupv2 bool) int64 {
+	cpuRequest := rr.Requests.Cpu()
+	cpuLimit := rr.Limits.Cpu()
+	var shares int64
+	if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+		shares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+	} else {
+		shares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+	}
+	if podOnCgroupv2 {
+		return 1 + ((shares-2)*9999)/262142
+
+	} else {
+		return shares
+	}
+}
+
+func getExpectedMemLimitString(rr *v1.ResourceRequirements, podOnCgroupv2 bool) string {
+	expectedMemLimitInBytes := rr.Limits.Memory().Value()
+	expectedMemLimitString := strconv.FormatInt(expectedMemLimitInBytes, 10)
+	if podOnCgroupv2 && expectedMemLimitString == "0" {
+		expectedMemLimitString = "max"
+	}
+	return expectedMemLimitString
+}
+
+func verifyCPUWeight(f *framework.Framework, pod *v1.Pod, containerName, cgPath string, expectedResources *v1.ResourceRequirements, podOnCgroupv2 bool) error {
+	cpuWeightCgPath := getCgroupCPURequestPath(cgPath, podOnCgroupv2)
+	expectedCPUShares := getExpectedCPUShares(expectedResources, podOnCgroupv2)
+	return VerifyCgroupValue(f, pod, containerName, cpuWeightCgPath, strconv.FormatInt(expectedCPUShares, 10))
+}
+
+func verifyCPULimit(f *framework.Framework, pod *v1.Pod, containerName string, cgPath string, expectedResources *v1.ResourceRequirements, podOnCgroupv2 bool) error {
+	cpuLimCgPath := getCgroupCPULimitPath(cgPath, podOnCgroupv2)
+	expectedCPULimits := getCPULimitCgroupExpectations(expectedResources.Limits.Cpu(), podOnCgroupv2)
+	return VerifyCgroupValue(f, pod, containerName, cpuLimCgPath, expectedCPULimits...)
+}
+
+func verifyMemoryLimit(f *framework.Framework, pod *v1.Pod, containerName, cgPath string, expectedResources *v1.ResourceRequirements, podOnCgroupv2 bool) error {
+	memLimCgPath := getCgroupMemLimitPath(cgPath, podOnCgroupv2)
+	expectedMemLim := getExpectedMemLimitString(expectedResources, podOnCgroupv2)
+	if expectedMemLim == "0" {
+		return nil
+	}
+	return VerifyCgroupValue(f, pod, containerName, memLimCgPath, expectedMemLim)
+}
+
+func verifyContainerCPUWeight(f *framework.Framework, pod *v1.Pod, containerName string, expectedResources *v1.ResourceRequirements, podOnCgroupv2 bool) error {
+	path := cgroupFsPath
+	if !podOnCgroupv2 {
+		path += "/cpu"
+	}
```

Comment:
Can this be moved into `getCgroupCPURequestPath`?

Reply:
I moved it to `getCgroupCPURequestPath` at the revision 2. I have been struggling to add subsystem to a path for both containers and pods on cgroup v1.